src/main/java/com/example/onlinebookstore/service/impl/CartItemServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.CartItemRequestDto;
+import com.example.onlinebookstore.dto.CartItemResponseDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.CartItemMapper;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.BookRepository;
+import com.example.onlinebookstore.repository.CartItemRepository;
+import com.example.onlinebookstore.service.CartItemService;
+import com.example.onlinebookstore.service.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CartItemServiceImpl implements CartItemService {
+    private final ShoppingCartService shoppingCartService;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+
+    @Override
+    @Transactional
+    public CartItemResponseDto add(CartItemRequestDto requestDto, User user) {
+        ShoppingCart shoppingCart = shoppingCartService.findByUserId(user);
+        CartItem cartItem = cartItemMapper.toEntity(requestDto);
+        Book book = bookRepository.findById(requestDto.getBookId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Book not found with id: " + requestDto.getBookId()));
+        cartItem.setBook(book);
+        cartItem.setShoppingCart(shoppingCart);
+        cartItemRepository.save(cartItem);
+        return cartItemMapper.toDto(cartItem);
+    }
+
+    @Override
+    @Transactional
+    public CartItemResponseDto updateCartItemQuantity(
+            Long id,
+            CartItemRequestDto requestDto,
+            User user
+    ) {
+        CartItem cartItem = cartItemRepository.findCartItemByBookId(requestDto.getBookId());
```

Comment:
this `findCartItemByBookId()` does not make much sense because different users can have the same book in their cart, and you do not have a way differentiate them in case there are several cart items with the same book id, `id` method parameter is also unused, so let's fetch the item directly by it's id, instead of book id

Reply:
Done